docs/topic-guides/deployment/aws-spot.txt

```diff
@@ -111,3 +111,27 @@ are not willing to delay their experiments. In this case, it may be best
 to not set ``spot_max_price`` and pay whatever the market price is. Your
 mileage may vary, but at Determined, we have regularly seen 70% cost
 reductions when using V100s, without specifying a ``spot_max_price``.
+
+*****************
+ Troubleshooting
+*****************
+
+Missing service-linked role
+===========================
+
+Some users may encounter the following error in the Determined Master
+logs the first time they try to use spot instances:
+
+.. code::
+
+   AWS error while launching spot instances, AuthFailure.ServiceLinkedRoleCreationNotPermitted, The provided credentials do not have permission to create the service-linked role for EC2 Spot Instances.
+
+When this error occurs, please check `AWS documentation
+<https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/spot-requests.html#service-linked-roles-spot-instance-requests>`__,
+or use AWS CLI to create the ``AWSServiceRoleForEC2Spot`` role:
```

Comment:
If this is always a reasonable solution, maybe just go to it directly:
```suggestion
When this error occurs, please use the AWS CLI to create the ``AWSServiceRoleForEC2Spot`` role:
```
and move the doc link to the "For more details..." line.

Reply:
I thought so at first (which is why there's a leftover "for more details" after the code), but then decided to do this way. Upon looking at in it in the UI, it felt a bit too harsh to tell people to do `aws iam` outright. It's permissions stuff, it may be a risk, so they should really read the docs and have some awareness.

Reply:
Ah, makes sense. The "or" sounds a little off and I kind of want them to be connected differently, though I'm not sure I can think of a good phrasing. Maybe:
```suggestion
When this error occurs, please check the `AWS documentation
<https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/spot-requests.html#service-linked-roles-spot-instance-requests>`__.
Most likely, you will need to use the AWS CLI to create the ``AWSServiceRoleForEC2Spot`` role:
```